documentation/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.md

```diff
@@ -70,8 +81,40 @@ value) in `AbstractMediaReport` definition will also need to be changed to
 `sensitive` and `sensitive_filtered` respectively. The migration necessary for
 changing the data is described below, however the API will need to continue to
 accept `mature` as a report reason and convert it internally to `sensitive` (at
-least until we decide to make a version change to the API). The media report
-tables already use table aliases as well
+least until we decide to make a version change to the API).
+
+In order to perform both of these changes in a
+[zero-downtime manner](https://docs.openverse.org/general/zero_downtime_database_management.html),
+the following steps will need to be taken:
+
+1. Rename the models (without changing the underlying tables) and associated
+   references for:
+   - `AbstractMatureMedia` -> `AbstractSensitiveMedia`
+   - `MatureImage` -> `SensitiveImage`
+   - `MatureAudio` -> `SensitiveAudio`
+2. Add `sensitive` and `sensitive_filtered` as a new status and reason
+   respectively for the `AbstractMediaReport` class. Also add
+   `sensitive_filtered` as to the check for
+   [creating sensitive media records](https://github.com/WordPress/openverse/blob/aa16d4f1be7607b12c428886b9890bdd947cc71c/api/api/models/media.py#L224).
+   Instance of `mature` that are supplied for the reporting endpoints should be
+   converted to `sensitive`.
+3. Add `sensitive` as an alias for the
+   [`mature` query parameter](https://github.com/WordPress/openverse/blob/3986e71c722e6cd8aca8a66a1f37d710aadc9a19/api/api/serializers/media_serializers.py#L120),
+   and deprecate the `mature` parameter.
```

Comment:
`mature` is already deprecated in favour of `include_sensitive_results`, so I don't think we would need to do this, right? Otherwise we're adding code (and complexity) to support a deprecated feature, which feels counter-intuitive to how I think of deprecated features :sweat_smile: